File: CSharp/TestExercises/Bank/Money.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TestExercises.Bank
{
    public class Money
    {
        private decimal value;

        public Money(int value) => this.value = value;

        public Money(decimal value) => this.value = value;

        public Money Add(Money other)
        {
            return new Money(value + other.value);
        }

        public Money Percentage(int p)
        {
            return new Money(value * p / 100);
        }

        public string Format()
        {
            return string.Format("{0:0.00}", value);
        }

        public decimal AsDecimal()
        {
            return value;
        }
    }
}

File: Java/src/main/java/at/nagarro/exercises/bank/Money.java
package at.nagarro.exercises.bank;

import java.math.BigDecimal;

public class Money {
    private BigDecimal value;

    public Money(double value) {
        this.value = new BigDecimal(value);
    }

    public Money(BigDecimal value) {
        this.value = value;
    }

    public BigDecimal getValue() {
        return value;
    }

    public Money add(Money other) {
        return new Money(value.add(other.value));
    }

    public Money percentage(int p) {
        return new Money(value.multiply(new BigDecimal(p / 100)));
    }

    public Money multiply(int multiplicand) {
        return new Money(this.getValue().multiply(new BigDecimal(multiplicand)));
    }

    public String format() {
        return String.format("{0:0.00}", value);
    }

}
